ui/src/components/Timeline2/util.ts

```diff
@@ -0,0 +1,257 @@
+import {
+  useMemo,
+  useCallback,
+  useEffect,
+  useState,
+  createRef,
+  KeyboardEvent,
+} from 'react';
+import { differenceInDays } from 'date-fns';
+import { throttle } from 'lodash';
+import { Entity, Schema } from '@alephdata/followthemoney';
+import { DEFAULT_COLOR } from './Timeline';
+import { FetchEntitySuggestions, Layout } from './types';
+
+function endOfMonth(year: number, month: number): number {
+  // Months in ECMAScript are zero-based:
+  // new Date(2000, 0, 1) // Jan 1st
+  //
+  // Setting the day to zero will return the last day of the previous month:
+  // new Date(2000, 1, 0) // Jan 31st
+  // new Date(2000, 12, 0) // Dec 31st
```

Comment:
True, this is because I’ve used the `/** ... */` style for JSDoc-like class comments, whereas `// ...` is used for comments describing implementation details. But I can change that to use `/* ... */` for those if you prefer.